app/(dashboard)/register/page.tsx

```diff
@@ -0,0 +1,496 @@
+"use client";
+
+import { api } from "@/convex/_generated/api";
+import { useMutation, useQuery } from "convex/react";
+import { useState, useEffect } from "react";
+import { useUser } from "@clerk/nextjs";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Badge } from "@/components/ui/badge";
+import { Checkbox } from "@/components/ui/checkbox";
+import { X, Plus } from "lucide-react";
+import { toast } from "sonner";
+import Link from "next/link";
+
+export default function Registration() {
+  const { user } = useUser();
+  const currentSubmission = useQuery(api.submission.getSubmission);
+  const submit = useMutation(api.submission.createOrUpdateSubmission);
+
+  const [formData, setFormData] = useState({
+    projectName: "",
+    members: [""],
+    githubUrl: "",
+    hostedSiteUrl: "",
+    videoOverviewUrl: "",
+    description: "",
+    favoriteParts: "",
+    biggestChallenges: "",
+    testingInstructions: "",
+    status: "in-progress" as "in-progress" | "submitted",
```

Comment:
this is `in-progress` if `currentSubmission` is null. Not a part of the form afaict?